includes/utilities.inc

```diff
@@ -232,3 +232,137 @@ function islandora_newspaper_get_date_issued(AbstractObject $object) {
   $date = empty($results) ? new DateTime() : new DateTime($results[0]['object']['value']);
   return $date;
 }
+
+/**
+ * Fetch the issues of a newspaper that do not have a date.
+ *
+ * @param AbstractObject|NULL $object
+ *   Newspaper object or empty for all issues
+ *
+ * @return array
+ *   Listing of the PIDs of objects without the date issued defined.
+ */
+function islandora_newspaper_get_empty_dates($object = NULL) {
+  $tuque = islandora_get_tuque_connection();
+
+  if (is_null($object)) {
+    $where = <<<EOQ
+  ?object <fedora-model:hasModel> <info:fedora/islandora:newspaperIssueCModel> ;
+EOQ;
+  }
+  else {
+    $where = <<<EOQ
+  ?object fedora-rels-ext:isMemberOf <info:fedora/{$object->id}> ;
+    <fedora-model:hasModel> <info:fedora/islandora:newspaperIssueCModel> ;
+EOQ;
+  }
+  $islandora_rels_ext = ISLANDORA_RELS_EXT_URI;
+  $fedora_rels_ext = FEDORA_RELS_EXT_URI;
+  $query = <<<EOQ
+PREFIX islandora-rels-ext: <{$islandora_rels_ext}>
+PREFIX fedora-rels-ext: <{$fedora_rels_ext}>
+SELECT ?object ?label
+FROM <#ri>
+WHERE {
+  $where
+    <fedora-model:label> ?label .
+  !optionals
+  !filters
+}
+ORDER BY ?label
+EOQ;
+
+  // XXX: Can't really use the fully featured query_filter and query_statements
+  // hook as it will return some things we don't want. Perhaps change the return
+  // structure in the future to specify which module they are coming from? For
+  // now we will just get XACML's directly.
+  $query_optionals = (array) module_invoke('islandora_xacml_api', 'islandora_basic_collection_get_query_optionals', 'view');
```

Comment:
i should have noticed this sooner but I think we should change view to manage in the call to islandora_basic_collection_get_query_optionals.  We want the user to be able to set the date so they will need manage access.